### PR TITLE
AB#14 dev tooling

### PR DESCRIPTION
**Pull request by mailaenderli**
- Updated trigger:
  - On PRs without integration tests
  - Merge into master with integration tests

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -6,6 +6,16 @@ variables:
...
 
 trigger:
+  batch: true
+  branches:
+    include:
+    - master
+  paths:
+    exclude:
+    - report/report.tex
+    - docs/README.md
```
Comment (StuckiSimon) [inline azure-pipelines.yml]: What's the purpose of the `docs/README.md` exclude? 🤔

File: azure-pipelines.yml [hunk 2]
```diff
@@ -33,6 +43,12 @@ steps:
   - script: |
       npm run test
     displayName: "test"
+
+  - script: |
+      npm run test-e2e:open
+      npm run test-e2e:run
```
Comment (StuckiSimon) [inline azure-pipelines.yml]: For the integration phase `test-e2e:run` would be sufficient. There is no need running `test-e2e:open` as well. Or am I being mistaken?

**Review by StuckiSimon — CHANGES_REQUESTED** [review]
Looks good, especially the trigger configuration and batching 👍 
But there are two open questions ✌️

File: azure-pipelines.yml (hunk 1 above)
Comment (mailaenderli) [reply azure-pipelines.yml]: That no build is triggered if only the report or readme changes

Comment (michaelschufi) [reply azure-pipelines.yml]: Makes sense, imo.

File: azure-pipelines.yml (hunk 2 above)
Comment (mailaenderli) [reply azure-pipelines.yml]: I thought that I need to Open a cypress "session" before running the tests. Should I remove the tests-e2e:open?

File: azure-pipelines.yml (hunk 1 above)
Comment (StuckiSimon) [reply azure-pipelines.yml]: Yeah sure, but it's README.md instead of docs/README.md, isn't it?

File: azure-pipelines.yml (hunk 2 above)
Comment (StuckiSimon) [reply azure-pipelines.yml]: Yeah, `test-e2e:run` will handle it for you - you can even test it locally ✌️

File: azure-pipelines.yml (hunk 1 above)
Comment (michaelschufi) [reply azure-pipelines.yml]: Oh, yes of course, sorry ^^

Comment (mailaenderli) [reply azure-pipelines.yml]: If we decide to leave this exclusions i need to add them to the pr section aswell

**Review by sir-marc — APPROVED** [review]

**Review by StuckiSimon — APPROVED** [review]

**Pull request merged**